### PR TITLE
Feature/get all movies fetch

### PR DESCRIPTION
**Pull request by wlavery22**
## Describe your changes
We switched from local data files to fetching data from an API, for displaying all movies, for displaying individual movie data, and added error handling.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [] If it is a core feature, I have added thorough tests.
- [x] There are no known bugs. If there are bugs, please list them below.

## Collaborators
Co-authored-by:  Matthew Shindel : <matthew@shindel.com>
Co-authored-by:  Dan Lavery : <lavery_william@yahoo.com>

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -45,9 +50,15 @@ function App() {
 	
   function fetchAllMovies() {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-    .then(response => response.json())
```
Comment (MatthewShindel) [inline src/components/App/App.js]: changed the .then() to make sure the response was ok. If the response wasn't ok, we'd throw an error, with the value of `${response.status}: ${response.statusText}`.

File: src/components/App/App.js [hunk 2]
```diff
@@ -70,26 +85,44 @@ function App() {
...
+				{/* {error && <h2>Error{error}</h2>} */}
       </header>
       <main>
-				{!showMovieInfo ? (
-					<AllMovies movies={movies} onMovieClick={handleMovieClick} updateMovieId={updateMovieId} updateMovieInfo={updateMovieInfo}/>
-				) : (
-					<MovieInfo onExitClick={handleBackClick} singleMovieInfo={singleMovieInfo} />
-				)
-				}
+		 {error ? (
```
Comment (MatthewShindel) [inline src/components/App/App.js:72]: Used a ternary for conditional rendering.

File: src/components/Movie/Movie.js [hunk 3]
```diff
@@ -37,4 +48,21 @@ function Movie({ id, posterPath, title, rating , onMovieClick, updateMovieId, up
 	)
 }
 
-export default Movie;
\ No newline at end of file
+export default Movie;
+
+function isURL(props, propName, componentName) {
```
Comment (MatthewShindel) [inline src/components/Movie/Movie.js:40]: Checks if the URL for an image(specifically the poster_path) is a valid image to use.

File: src/components/Movie/Movie.js [hunk 4]
```diff
@@ -37,4 +48,21 @@ function Movie({ id, posterPath, title, rating , onMovieClick, updateMovieId, up
...
+  if (!/^https?:\/\/\S+$/.test(props[propName])) {
+    return new Error(
+      `Invalid prop ${propName} supplied to ${componentName}. Validation failed.`
+    );
+  }
+}
+
+Movie.propTypes = {
+  id: PropTypes.number.isRequired,
```
Comment (MatthewShindel) [inline src/components/Movie/Movie.js:49]: data validation for our movie.props

**Comment by MatthewShindel** [comment]
Overall, great work, good job using Proptypes.

File: src/components/AllMovies/AllMovies.js [hunk 5]
```diff
@@ -27,4 +29,12 @@ function AllMovies( {movies , onMovieClick} ){
 	)
 }
 
-export default AllMovies;
\ No newline at end of file
+export default AllMovies;
+
+AllMovies.propTypes = {
+  movies: PropTypes.array.isRequired,
```
Comment (nicktu12) [inline src/components/AllMovies/AllMovies.js:35]: PropTypes on complex data types (arrays, objects) need to be deeply validated. What is the data types of the elements inside the array/object?

File: src/components/App/App.js [hunk 6]
```diff
@@ -9,41 +8,74 @@ function App() {
...
+  const [error, setError] = useState("")
+  
 
-	const handleMovieClick = () => {
+  function handleMovieClick() {
 		setShowMovieInfo(true);
-		console.log("hit handleMovieClick");
+  }
+
+	function handleError(error) {
```
Comment (nicktu12) [inline src/components/App/App.js:20]: are these three wrapper functions needed? Maybe for passing props? They all look super similar, and they dont really do anything besides calling another function.

File: src/components/Movie/Movie.js [hunk 7]
```diff
@@ -1,26 +1,55 @@
...
 		</div>
 	)
 }
 
-export default Movie;
\ No newline at end of file
+export default Movie;
+
+function isURL(props, propName, componentName) {
+  if (!/^https?:\/\/\S+$/.test(props[propName])) {
```
Comment (nicktu12) [inline src/components/Movie/Movie.js:41]: Regex!! Nice. I would make sure you have a high level understanding of whats going on in this regular expression

**Pull request merged**